src/async/retry.ts

```diff
@@ -3,16 +3,24 @@ import { sleep, tryit } from 'radashi'
 export type RetryOptions = {
   times?: number
   delay?: number | null
+  onRetry?: (err: Error, attemptNumber: number) => void
   backoff?: (count: number) => number
 }
 
 /**
  * Retries the given function the specified number of times.
  *
+ * @param {Object} options - The employee who is responsible for the project.
+ * @param {number} [options.times=3] - Number of attempts.
+ * @param {number} [options.delay] - Specify milliseconds to sleep between attempts.
+ * @param {Function} [options.onRetry=null] - Is invoked after a new retry is performed. It's passed the Error that triggered it as a parameter and the attempt number.
+ * @param {Function} [options.backoff=null] - The backoff option is like delay but uses a function to sleep -- makes for easy exponential backoff.
+ * @param {Function} func - Function to be executed
+ *
  * @see https://radashi.js.org/reference/async/retry
  * @example
  * ```ts
```

Comment:
when i said to use Jsoc is in the Type, because using only in the function we does not have a good integration with VSCode, see: 

after: 
![image](https://github.com/user-attachments/assets/810a89f4-c0bb-490a-b6db-55e1654f2680)

before: 
![image](https://github.com/user-attachments/assets/e364e446-d65e-4fc5-a054-33b4a137f0b7)

```suggestion
export type RetryOptions = {
  /**
   * Number of attempts.
   *
   * @default 3
   */
  times?: number
  /** Specify milliseconds to sleep between attempts */
  delay?: number | null
  /**
   * Is invoked after a new retry is performed. It's passed the Error that triggered it as a parameter and the attempt number.
   *
   * @default null
   */
  onRetry?: (err: Error, attemptNumber: number) => void
  /**
   * The backoff option is like delay but uses a function to sleep -- makes for easy exponential backoff.
   *
   * @default null
   */
  backoff?: (count: number) => number
}

/**
 * Retries the given function the specified number of times.
 *
 * @see https://radashi.js.org/reference/async/retry
 * @example
 * ```ts
```